src/time.rs

```diff
@@ -0,0 +1,39 @@
+use std::{
+    cell::RefCell,
+    collections::hash_map::{Entry, HashMap},
+    ffi::OsString,
+    time::Instant,
+};
+
+thread_local!( static INSTANTS: RefCell<HashMap<OsString, Instant>> = RefCell::new(HashMap::new()) );
```

Comment:
Using OsString here is likely unnecessary; BYOND has also standardized on UTF-8, just like Rust Strings.

Reply:
Ah, honestly I just loked at whatever `log.rs` used I think because I wasn't sure.